src/lsm/lsm_cursor.c

```diff
@@ -493,9 +532,7 @@ retry:	if (F_ISSET(clsm, WT_CLSM_MERGE)) {
 		 * to detect snapshot isolation conflicts.
 		 */
 		if (F_ISSET(clsm, WT_CLSM_OPEN_SNAPSHOT))
-			WT_ERR(__wt_realloc_def(session,
-			    &clsm->txnid_alloc, nchunks,
-			    &clsm->switch_txn));
+			WT_ERR(__clsm_alloc_iter(session, clsm, nchunks));
```

Comment:
Did you consider making this unconditional, so we always allocate the full chunk array here?

We could move the other call at line 594 up into the `WT_CLSM_MERGE` case rather than calling it twice in the common case?